### PR TITLE
Language selection

### PR DESCRIPTION
**Pull request by laurensmartina**

### CODE REVIEW

File: system/core/language.php [hunk 1]
```diff
@@ -24,9 +24,16 @@ function __($msgid, $args = null) {
...
 	function languageOptionList($select, $omit) {
 		global $isoLangList;
-		$html = '';
-		foreach($isoLangList as $code => $langName) if ($code!=$omit) $html.= '<option value='.$code.($code==$select ? ' selected' : '').'>'.$code.': '.$langName.'</option>';
-		return addslashes($html);
+		$langList = hypha_getUsedContentLanguages();
+
+		$inUse = '';
+		foreach ($langList as $code) if ($code!=$omit) $inUse.= '<option value='.$code.($code==$select ? ' selected' : '').'>'.$code.': '.$isoLangList[$code].'</option>';
+		$html = '<optgroup label="in use">' . addslashes($inUse) . '</optgroup>';
```
Comment (matthijskooijman) [inline system/core/language.php]: This is probably a good moment to add html escaping, and remove this pointless addslashes (probably in a subsequent commit, though).

File: system/core/pages.php [hunk 2]
```diff
@@ -100,6 +100,7 @@ function newPage() {
 		html+= '<tr><th><?=__('type')?></th><td><select id="newPageType" name="newPageType">' + '<?php foreach($types as $type => $datatypeName) echo '<option value="'.$type.'"'.($type=='textpage' ? 'selected="selected"' : '').'>'.$datatypeName.'</option>'; ?>' + '</select> ' + infoPageType + '</td></tr>';
 		var infoPageName = <?=json_encode(makeInfoButton('help-page-name'));?>;
 		html+= '<tr><th><?=__('pagename')?></th><td><input type="text" id="newPagename" value="<?=$pagename?>" onblur="validatePagename(this);" onkeyup="validatePagename(this); document.getElementById(\'newPageSubmit\').disabled = this.value ? false : true;"/> ' + infoPageName + '</td></tr>';
+		html+= '<tr><th><?=__('language')?></th><td><select id="newPageLanguage" name="newPageLanguage"><?=languageOptionList('', '')?></select></td></tr>';
```
Comment (matthijskooijman) [inline system/core/pages.php]: Should this maybe preselect the default or current language? Now, nothing is marked as selected, so it just defaults to the first element in the list.

File: system/core/pages.php [hunk 3]
```diff
@@ -276,18 +277,18 @@ function loadPage(RequestContext $O_O) {
 	registerCommandCallback('newPage', 'newPage');
 	function newPage($newName) {
 		global $hyphaXml, $hyphaUrl, $hyphaContentLanguage;
-
+		$language = array_key_exists('newPageLanguage', $_POST) ? $_POST['newPageLanguage'] : $hyphaContentLanguage;
```
Comment (matthijskooijman) [inline system/core/pages.php]: I think this should validate the language code, since AFAICS `hypha_addPage` just uses it as-is.

File: index.php [hunk 4]
```diff
@@ -108,6 +108,10 @@
 	 * @deprecated Use O_O instead
 	 */
 	$hyphaLanguage = $O_O->getInterfaceLanguage();
+	$dict = Language::getDictionaryByLanguage($hyphaLanguage);
+	if (array_key_exists('_locale', $dict)) {
```
Comment (matthijskooijman) [inline index.php]: Why not use `__()` here? The only caveat I can think of is that it would return `_locale` if not found, but that seems easy to check?

**Comment by matthijskooijman** [comment]
@laurensmartina I see you pushed to this PR, but it seems my comments are still unfixed. Did you just rebase on top of master maybe?

File: system/core/language.php (hunk 1 above)
Comment (laurensmartina) [reply system/core/language.php]: Tested it, and removing the `addslashes` results in errors.

File: system/core/pages.php (hunk 2 above)
Comment (laurensmartina) [reply system/core/pages.php]: fixed

File: index.php (hunk 4 above)
Comment (laurensmartina) [reply index.php]: fixed

File: system/core/pages.php (hunk 3 above)
Comment (laurensmartina) [reply system/core/pages.php]: fixed at `hypha_addPage `

File: system/core/language.php (hunk 1 above)
Comment (matthijskooijman) [reply system/core/language.php]: I've removed the addSlashes and added htmlspecialchars, just force pushed this.

**Comment by matthijskooijman** [comment]
I just rebased this on master and added a few more commits (one at the end, and a handful at the beginning). @laurensmartina, could you review my new commits? If they're ok, I think this is ready to be merged.

File: system/php-dom-wrapper/Traits/ManipulationTrait.php [hunk 5]
```diff
@@ -220,6 +220,8 @@ public function setText($input) {
             // Add new contents in it's place.
             $node->append(new Text($newNodes->getText()));
         });
+
+        return $this;
```
Comment (laurensmartina) [inline system/php-dom-wrapper/Traits/ManipulationTrait.php]: There is an inconsistency with the return type in the annotation and the actual return type. Please make it consistent.

**Comment by matthijskooijman** [comment]
Thanks for the review. I've fixed the comment, will merge this now.

**Pull request merged**